src/lib.rs

```diff
@@ -77,7 +88,27 @@ fn approx_run(
                 }
                 #[cfg(feature = "mpi")]
                 Parallelism::MPI => {
-                    christofides::<{ computation_mode::MPI_COMPUTATION }>(&tsp_instance.graph)
+                    let universe = mpi::initialize().unwrap();
+                    let world = universe.world();
+                    let rank = world.rank();
+                    let root_process = world.process_at_rank(0);
+
+                    if rank == ROOT_RANK {
```

Comment:
It took me a bit to understand this code but... does this mean that christofides deadlocks whenever not called from this lib, as the else is expected to also be called on other ranks?

If thats the case, and christofides is a `pub fn` (as it should), we should refactor that in the future.
For now, as we have a lot of time constraints, I'd be fine if we merge, you write an issue explaining it and annotate the issue as a TODO comment to this code.

If you need help writing this issue, just hmu

Reply:
Yeah, that might be a problem. That truly is something to think about later.
My goal was to keep the uniform function signature across sequential, rayon, and mpi mode.
Otherwise it would suffice to pass a `world` object to the function in mpi mode